web/store/page.store.ts

```diff
@@ -184,15 +184,21 @@ export class PageStore implements IPageStore {
    * @param projectId
    * @returns Promise<IPage[]>
    */
-  fetchProjectPages = async (workspaceSlug: string, projectId: string) =>
-    await this.pageService.getProjectPages(workspaceSlug, projectId).then((response) => {
-      runInAction(() => {
-        response.forEach((page) => {
-          set(this.pages, [page.id], page);
+  fetchProjectPages = async (workspaceSlug: string, projectId: string) => {
+    try {
+      return await this.pageService.getProjectPages(workspaceSlug, projectId).then((response) => {
+        console.log("Response from backend 1", response);
+        runInAction(() => {
+          response.forEach((page) => {
+            set(this.pages, [page.id], page);
+          });
         });
+        return response;
       });
-      return response;
-    });
+    } catch (error) {
+      return [];
```

Comment:
catch block should always throw the error message, to handle it at the component level. By returning an empty array we can never identify if the server value is empty or it is an error case.